### PR TITLE
LPS-112699 Revert type change because other places in Forms assume this should be a List

### PR DESCRIPTION
**Pull request by brunobasto**

### CODE REVIEW

**Comment by brunobasto** [comment]
ci:test:sf

**Comment by brunobasto** [comment]
ci:test:relevant

**Comment by brunobasto** [comment]
ci:test:stable

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/fieldset/FieldSetDDMFormFieldTemplateContextContributor.java [hunk 1]
```diff
@@ -78,25 +78,55 @@
...
+			columnJSONObject.put(
+				"fields", fieldsJSONArray
+			).put(
+				"size", columnSize
+			);
+
+			columnsJSONArray.put(columnJSONObject);
+		}
+
+		JSONObject rowJSONObject = jsonFactory.createJSONObject();
```
Comment (leoadb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/fieldset/FieldSetDDMFormFieldTemplateContextContributor.java:123]: Use JSONUtil instead

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/fieldset/FieldSetDDMFormFieldTemplateContextContributor.java [hunk 2]
```diff
@@ -78,25 +78,55 @@
...
+		JSONArray columnsJSONArray = jsonFactory.createJSONArray();
+
+		for (Object nestedFieldContext : nestedFields) {
+			JSONArray fieldsJSONArray = jsonFactory.createJSONArray();
+
+			fieldsJSONArray.put(
+				MapUtil.getString(
+					(Map<String, ?>)nestedFieldContext, "fieldName"));
+
+			JSONObject columnJSONObject = jsonFactory.createJSONObject();
```
Comment (leoadb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/fieldset/FieldSetDDMFormFieldTemplateContextContributor.java:112]: Use JSONUtil instead

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/fieldset/FieldSetDDMFormFieldTemplateContextContributor.java [hunk 3]
```diff
@@ -78,25 +78,55 @@
...
+
+			columnJSONObject.put(
+				"fields", fieldsJSONArray
+			).put(
+				"size", columnSize
+			);
+
+			columnsJSONArray.put(columnJSONObject);
```
Comment (leoadb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/fieldset/FieldSetDDMFormFieldTemplateContextContributor.java:120]: Add the return from JSONUtil inline

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/fieldset/FieldSetDDMFormFieldTemplateContextContributor.java [hunk 4]
```diff
@@ -78,25 +78,55 @@
...
+			fieldsJSONArray.put(
+				MapUtil.getString(
+					(Map<String, ?>)nestedFieldContext, "fieldName"));
+
+			JSONObject columnJSONObject = jsonFactory.createJSONObject();
+
+			columnJSONObject.put(
+				"fields", fieldsJSONArray
+			).put(
+				"size", columnSize
```
Comment (leoadb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/fieldset/FieldSetDDMFormFieldTemplateContextContributor.java:117]: Don't create the variable columnSize, add the expression inline instead

Comment (brunobasto) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/fieldset/FieldSetDDMFormFieldTemplateContextContributor.java:117]: even inside a loop?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/fieldset/FieldSetDDMFormFieldTemplateContextContributor.java [hunk 5]
```diff
@@ -127,26 +157,34 @@ protected JSONArray getJSONArray(String rows) {
...
 
-		if (Validator.isNotNull(nestedFieldNames)) {
-			return StringUtil.split(nestedFieldNames);
-		}
+		Stream<Object> visibleNestedFieldsStream = nestedFields.stream();
 
-		return defaultNestedFieldNames.toArray(new String[0]);
-	}
+		List<Object> visibleNestedFields = visibleNestedFieldsStream.filter(
+			nestedField -> isNestedFieldVisible(nestedField)
```
Comment (leoadb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/fieldset/FieldSetDDMFormFieldTemplateContextContributor.java:166]: Replace with **this::isNestedFieldVisible**

**Comment by brunobasto** [comment]
ci:forward

**Pull request closed without merging**